hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/operations/GenericOperation.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.ringbuffer.impl.operations;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.ringbuffer.impl.RingbufferContainer;
+
+import java.io.IOException;
+
+import static com.hazelcast.ringbuffer.impl.RingbufferDataSerializerHook.GENERIC_OPERATION;
+
+public class GenericOperation extends AbstractRingBufferOperation {
+
+    public static final byte OPERATION_SIZE = 0;
```

Comment:
What about using an enum here? Ordinal values can be written to output with DataOutput.writeByte(int).